src/app/lamp_core/base_lamp.py

```diff
@@ -0,0 +1,30 @@
+from ..network.coding import Codes
+from ..network.network import LampNetworkObserver
+
+class BaseLamp(LampNetworkObserver):
+    '''
+    Trival Lamp framework that real lamps will be based on.
+    Provides a simple interface for the rest of the system to build
+    up the lamp and does basic broadcasting about the lamp being created.
+    '''
+    def __init__(self, network, base_color, shade_color):
```

Comment:
where does `network` come from?

Reply:
`network` is created in the `App` layer and injected to all the lamps.